### PR TITLE
Misc type annotations

### PR DESCRIPTION
**Pull request by Carreau**
Extracted from #1272

### CODE REVIEW

File: ipykernel/kernelapp.py [hunk 1]
```diff
@@ -142,9 +143,10 @@ class IPKernelApp(BaseIPythonApplication, InteractiveShellApp, ConnectionFileMix
     debug_shell_socket = Any()
     stdin_socket = Any()
     iopub_socket = Any()
-    iopub_thread = Any()
-    control_thread = Any()
-    shell_channel_thread = Any()
+
+    iopub_thread: BaseThread
```
Comment (minrk) [inline ipykernel/kernelapp.py:147]: I think these still need to be defined as traits, e.g. `: BaseThread = Any()` or perhaps even `Instance(BaseThread, allow_none=True)`.

Comment (Carreau) [reply ipykernel/kernelapp.py:147]: I think that will need a `# type: ignore[assign]` as well, or mypy will be unhappy

**Comment by Carreau** [comment]
Updated, and fixed the corresponding error shown by mypy.

**Comment by minrk** [comment]
Thanks!

**Comment by Carreau** [comment]
thanks for the review and merge !

**Pull request merged**